### ISSUE TITLE
Crafting recipe for written book is not shown

### ISSUE THREAD

**Issue by Wuzzy2** [minetest-mods/craftguide#16]
The following shapeless recipe from Minetest Game to craft a “Book With Text” is *not* shown in the crafting guide:

    Book With Text (`default:book_written`) + Book (`default:book`)

This is a spcial recipe to copy the contents of the book with text into the book.
This recipe clearly needs to be shown in the crafting guide.

I also know the reason. In `craftguide:get_init_items`, the function filters out items with `not_in_creative_inventory=1`. This is the case for “Book With Text”.

I think this check should be removed, but I can also see the intention here: To reduce possible recipe clutter. However, I think there could be a better way to hide certain recipes.

I suggest to check not against `not_in_creative_inventory`, but against `not_in_craft_guide`. This group is already being used by `zcg` and `craft_guide`, so it kinda looks like an informal standard. Maybe `craftguide` should just adopt it. :-)

**Comment by kilbith** [comment]
Yes, I can't remove the check for `not_in_creative_inventory=1` because e.g. `moreblocks` registers thousands of recipes for the `stairplus` nodes (all the circular saw's output).

So I can't replace this group by `not_in_craft_guide` unless if all the mainstream mods decide to add to it at appropriate definitions.

However we could add an exception for the book with text, but IMO it's inappropriate to show it in the Crafting Guide. It wouldn't explain what this craft does exactly (i.e. content reproduction).

**Comment by Wuzzy2** [comment]
OK, at least you could add an additional check for `not_in_craft_guide=1` to make this group more of a de facto standard. I think it doesn't hurt.

I would still include the recipe for the book with text, even if it is not clear why it exists. A curious player might just try it out. ;-)
Too bad there is no real automated way to figure out whether a recipe is “special”, i.e. something special happens when you craft it.

**Comment by FaceDeer** [comment]
Maybe a more practical approach could be to add an "include_in_craft_guide" group that overrides the not_in_creative_inventory check, and perhaps some other optional overrides to allow special descriptions (such as "copies text from one book to another"),

That would allow special cases like written books to explicitly tell the crafting guide what to do about them, while leaving the default case alone and not requiring any updates to unrelated mods.

**Comment by dsohler** [comment]
:+1: for adding `not_in_craft_guide` as second check for loading the recipes.

**Comment by pauloue** [comment]
How does the latest commit change this? The recipe for written book is still not shown.
